action.yml

```diff
@@ -24,5 +28,30 @@ inputs:
     default: 'gpt-4-turbo-preview'
 
 runs:
-  using: 'docker'
-  image: 'Dockerfile' 
\ No newline at end of file
+  using: 'composite'
+  steps:
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+        cache: 'pip'
+    
+    - name: Install dependencies
+      shell: bash
```

Comment:
📦 The 'Install dependencies' step has been added, which is crucial for the action's execution. Verify that the installed dependencies are up-to-date and necessary for the action's functionality.